src/showlib/l1b/l1b.py

```diff
@@ -0,0 +1,368 @@
+
+    def phase(self, signal, freq, deg):
+        return np.polyfit(freq[:], np.unwrap(np.angle(signal[:])), deg=deg)
+
+    def get_full_spectrum(self, sample_interval, data: np.ndarray, pad_factor):
+        """ "Take the FFT of each row in the image and return the full complex spectrum and spatial frequency scale
+
+        Parameters
+        ----------
+        sample_interval: float
+            sample spacing
+        data: np.ndarray
+            array containing the interferogram image
+        Returns
+        -------
+        s: np.ndarray
+            complex spectrum
+        f: np.ndarray
+            spatial frequencies
+
+        """
+
+        data = np.pad(data, pad_factor * 1000)
+        N = len(data)
+        delta = sample_interval
+
+        x = np.fft.ifft(np.fft.ifftshift(data))
+        s = np.fft.fftshift(x)
+
+        if N % 2 == 0:
+            n = np.arange(-int(N / 2), int(N / 2), 1)
+        else:
+            # n = np.arange(-int((N - 1) / 2), int((N - 1) / 2), 1)
+            n = np.arange(-int((N - 1) / 2), int((N - 1) / 2 + 1), 1)
+        f = n / (N * delta)
+        return s, f
+
+    def phase_correction(self, iGM):
+        # make a box car function centered at the center of the signal with a width that is half the number of samples
+        # slide the box car along the interferogram near the known center to find the location that minimizes the imaginary component
+        array_center = int(len(iGM) / 2)
+        d = []
+        idx_list = np.arange(-5, 5)
+        fft = []
+        phase = []
+        igms = []
+        for i in idx_list:
+            b = self.box_car(len(iGM), int(len(iGM) / 2) - 10, c=array_center + i)
+            iGMc = b * iGM
+            iGM_padded = []
+            # pad the edge of the interferogram with zeros to keep it symmetric
+            if i < 0:
+                iGM_padded.append(np.pad(iGMc, (2 * np.abs(i), 0)))
+
+            if i == 0:
+                iGM_padded.append(iGMc)
+
+            if i > 0:
+                iGM_padded.append(np.pad(iGMc, (0, 2 * i)))
+
+            igms.append(iGM_padded)
+            iGM_fft, f_fft = self.get_full_spectrum(
+                self.specs["opd_spacing"], data=iGM_padded[0], pad_factor=0
+            )
+            d.append(np.mean(np.diff(np.unwrap(np.angle(iGM_fft[100:160])))))
+            phase.append(np.unwrap(np.angle(iGM_fft[100:160])))
+            fft.append(iGM_fft)
+
+        idx_center = np.argmin(np.abs(d))
+        iGM_fft_center = fft[idx_center]
+        n = len(iGM_fft_center)
+        fx = (2 * np.pi / (n * np.mean(np.diff(self.specs["pos_x"])))) * np.arange(
+            -(n - 1) / 2, (n - 1) / 2 + 1
+        )
+        kL = 8 * np.pi * np.tan(self.specs["ThetaL"]) * self.specs["Littrow"]
+
+        lstsq_fit = least_squares(
+            self.minfunc,
+            (1e-05, 1),
+            args=(fx[120:160], iGM_fft_center[120:160]),
```

Comment:
can these bounds bo defined relative to `n`?

Reply:
Corrected